azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java

```diff
@@ -372,14 +376,43 @@ public void uploadProjectFile(Project project, int version, String filetype,
     }
   }
 
-  private void uploadProjectFile(Connection connection, Project project,
-      int version, String filetype, String filename, File localFile,
+  private void uploadProjectFile(Connection connection, int projectId, int version, File localFile)
```

Comment:
Understand that we need to do version initialization in a separate metho, but can't see the benefits why we need a separate method to have the two steps below